scripts/secondary.py

```
@@ -25,7 +25,7 @@
                     if 'c' in ns and ns['c'] == localhost['dns_facts_secondary_name']:
                         ret[name] = {
                             'kind': 'Slave',
-                            'masters': [hostvars[hostname]['ansible_host']]
+                            'masters': [hostvars[hostname]['dns_facts_dns_ipv4']]
```

Maybe we can default to `ansible_host` when `dns_facts_dns_ipv4` is not set?

---

That makes it possible to accidentally use `ansible_host` instead of the correct IP. That's what happened to us over the last days, leading to our dns02 not being able to sync with dns01.

---

Also, `ansible_host` doesn't strictly need to be an IP. It can also be a hostname or DNS name. It's just an oddity of our setup that we expect at multiple places that `ansible_host` is an IPv4 address.

---

Hm ok. Can we make it fail, when the variable is not set?

---

> This variable is required to set on primary servers that have at least one secondary.

If this is not satisfied, then it does fail when deploying the secondary.

---

ok